lfsapi/ssh.go

```diff
@@ -100,6 +101,12 @@ func (c *sshAuthClient) Resolve(e Endpoint, method string) (sshAuthResponse, err
 		res.Message = strings.TrimSpace(errbuf.String())
 	} else {
 		err = json.Unmarshal(outbuf.Bytes(), &res)
+		if res.ExpiresIn <= 0 {
```

Comment:
I think this line has the potential to cause issues when the server _does_ send the `ExpiresIn` property, but it is equal to 0, which the spec permits [[1]](https://github.com/git-lfs/git-lfs/blob/master/tq/schemas/http-batch-response-schema.json#L17-L21).

I believe what will happen here is that the token is expired, this change will instruct LFS to fallback to `lfs.defaulttokenttl`, and the request will be made in error.

Do you think that we could change this to `if res.ExpiresIn != 0`, since the server is allowed to send negative values (indicating that the token has already expired?) `res.ExpiresIn == 0` will still be broken, but that's OK since there's no way to distinguish between `0` and "I didn't send you anything".

Reply:
I think the best approach might be to change ExpiresIn and ExpiresAt to pointers as it unmarshals pointers correctly and we need to know if they exist or not. I've updated the pull request addressing that.

Do you have any objections to adding an environment variable in addition to the configuration option?